### ISSUE TITLE
syntax highlighting doesn't understand heredocs

### ISSUE THREAD

**Issue by masukomi** [Raku/raku-mode#58]
The contents of a heredoc are highlighted as if they were code, instead of a string. 
See attached screenshot.

<img width="1126" alt="screenshot showing heredoc highlighting in emacs" src="https://user-images.githubusercontent.com/2829/187207355-ca31f06d-e522-4887-aae7-44646f5e8823.png">

**Comment by tbrowder** [comment]
Maybe so, but I wish an emacs dev could fix it.

**Comment by finanalyst** [comment]
There is a big problem with syntax highlighting of Raku in general because it has slangs, or sub languages. The slangs are Pod, regex, and quoting. So, a syntax highlighter developer will tend to chose a single way of colouring all of a Pod/regex/quoting section.
The two examples you have given above are different in that text within a single quote is obviously a string, which is coloured green.
The first is a quoted text, which the highlighter treats as code. This may be a reasonable choice for say `qq` which interpolates or `qx` which can execute.
Another problem I have found with syntax hightlighting that no none raku highlighter manages is the ending of a line with a quote, or double quotes, eg
```
say qq:to/MSG/ if True;
stuff in message
MSG
say "two messages" ~ q:to/MSG1/ ~ "and secondly " ~ q:to/MSG2/;
This is message one
MSG1
This is message two
MSG2
```
Similar sorts of problems exist with POD and regex, which need their own colouring rules.
Raku is a bit beyond most other languages.

**Comment by masukomi** [comment]
>  Raku is a bit beyond most other languages.

this is true, but  with regards to heredocs specifically: They're not notably different from Ruby's heredocs which basically every editor handles by just making all the contents be highlighted like any other string regardless of it they can be interpolated or not. 

I don't see `q:to/END/;...END` as being notably different for the purposes of highlighting form `<<~END...END`

With regards to your example of embedded heredocs, as a _human_ i'm not even sure what the "right" way to highlight that is so i wouldn't complain about an editor's highlighter saying "screw it. it's a comment" 

examples of ruby heredocs. Ruby also has some new(ish) funky one where you start it in a method call but finish it after it, but i forget how to do that. 

emacs highlights the following heredocs pretty similarily to what github is doing.

I feel like "perfect" isn't required here. I think a simple "just treat it like a string" is fine. 

```ruby
    type  = "healthy"
    table = "food"

    query = <<-SQL
SELECT * FROM #{table}
WHERE #{type} = true
SQL

    query = <<~SQL
    SELECT * FROM #{table}
    WHERE #{type} = true
    SQL

```
